homeassistant/components/daikin/__init__.py

```diff
@@ -4,112 +4,105 @@
-from homeassistant.const import (
-    CONF_HOSTS, CONF_ICON, CONF_MONITORED_CONDITIONS, CONF_NAME, CONF_TYPE
-)
-from homeassistant.helpers import discovery
-from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.typing import HomeAssistantType
 from homeassistant.util import Throttle
 
+from . import config_flow  # noqa  pylint_disable=unused-import
+from .const import KEY_HOST
+
 REQUIREMENTS = ['pydaikin==0.8']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'daikin'
 
-ATTR_TARGET_TEMPERATURE = 'target_temperature'
-ATTR_INSIDE_TEMPERATURE = 'inside_temperature'
-ATTR_OUTSIDE_TEMPERATURE = 'outside_temperature'
 
 MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
 
 COMPONENT_TYPES = ['climate', 'sensor']
 
-SENSOR_TYPE_TEMPERATURE = 'temperature'
-
-SENSOR_TYPES = {
-    ATTR_INSIDE_TEMPERATURE: {
-        CONF_NAME: 'Inside Temperature',
-        CONF_ICON: 'mdi:thermometer',
-        CONF_TYPE: SENSOR_TYPE_TEMPERATURE
-    },
-    ATTR_OUTSIDE_TEMPERATURE: {
-        CONF_NAME: 'Outside Temperature',
-        CONF_ICON: 'mdi:thermometer',
-        CONF_TYPE: SENSOR_TYPE_TEMPERATURE
-    }
-
-}
-
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Optional(
             CONF_HOSTS, default=[]
         ): vol.All(cv.ensure_list, [cv.string]),
-        vol.Optional(
-            CONF_MONITORED_CONDITIONS,
-            default=list(SENSOR_TYPES.keys())
-        ): vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
     })
 }, extra=vol.ALLOW_EXTRA)
 
 
 def setup(hass, config):
     """Establish connection with Daikin."""
-    def discovery_dispatch(service, discovery_info):
-        """Dispatcher for Daikin discovery events."""
-        host = discovery_info.get('ip')
-
-        if daikin_api_setup(hass, host) is None:
-            return
-
-        for component in COMPONENT_TYPES:
-            load_platform(hass, component, DOMAIN, discovery_info,
-                          config)
+    if DOMAIN not in config:
+        return True
+
+    hosts = config[DOMAIN].get(CONF_HOSTS)
+    if not hosts:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={'source': config.SOURCE_IMPORT}))
+    for host in hosts:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={'source': config.SOURCE_IMPORT},
+                data={
+                    KEY_HOST: host,
+                }))
+    return True
 
-    discovery.listen(hass, SERVICE_DAIKIN, discovery_dispatch)
 
-    for host in config.get(DOMAIN, {}).get(CONF_HOSTS, []):
-        if daikin_api_setup(hass, host) is None:
-            continue
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry):
+    """Establish connection with Daikin."""
+    conf = entry.data
+    daikin_api = await daikin_api_setup(hass, conf[KEY_HOST])
+    if not daikin_api:
+        return
```

Comment:
Here we should `return False`, since we `return True` if the logic flow is positive.

Reply:
Return values should be consistent and match. If we return a boolean at some point in a function, all return values of that function should be boolean.